src/axes/axis.ts

```diff
@@ -288,13 +290,33 @@ export class Axis<D> extends Component {
    */
   public tickLength(length: number): Axis<D>;
   public tickLength(length?: number): any {
+    Utils.Window.deprecated("tickLength()", "v1.2.0", "Renamed to innerTickLength()");
+    return this._tickLength(length);
```

Comment:
Since this is the same as saying `return this.innerTickLength(length)`, how about we transform `_tickLength()` into `innerTickLength()`. 

In other words, have just `tickLength()` and `innerTickLength`. The former will call `deprecated` and then `innerTickLength`.